### PR TITLE
Skip last pull if nothing to push

### PR DESCRIPTION
**Pull request by leplatrem**
@Natim @n1k0 r?

### CODE REVIEW

File: test/collection_test.js [hunk 1]
```diff
@@ -894,6 +894,17 @@ describe("Collection", () => {
...
     });
 
+    it("should not execute a last pull if nothing to push", (done) => {
+      const result = new SyncResultObject();
+      sandbox.stub(articles, "pullChanges").returns(Promise.resolve(result));
+      articles.clear()
+        .then(_ => articles.sync())
+        .then(_ => {
+          sinon.assert.calledOnce(articles.pullChanges);
+          done();
```
Comment (n1k0) [inline test/collection_test.js]: Nit: If you return the promise, you don't need to use `done` at all.

**Comment by n1k0** [comment]
r+ with failing test updated to reflect the change (s/calledTwice/calledOnce)

**Comment by leplatrem** [comment]
Thanks !
Rewrote test.

**Comment by Natim** [comment]
r+

**Pull request merged**